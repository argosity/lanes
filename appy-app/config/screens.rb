
Lanes::Screen.define "base" do | screen |
    screen.description = "Base Screen for Appy App"
    screen.icon        = ""
    screen.group_id    = ""
    screen.model_class = ""
    screen.view_class  = "AppyApp.Screens.Base"
end
