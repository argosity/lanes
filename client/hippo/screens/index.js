import { observable, computed } from 'mobx';
import { map, filter, intersection } from 'lodash';
import Group from './group';
import Config from '../config';

const Screens = observable({

    all: observable.map(),

    get activeGroups() {
        const gids = Group.enabled_group_ids || map(Group.all, 'id');
        const groups = map(gids, gid => Group.forId(gid));
        return filter(groups, group => intersection(group.screens, this.active).length);
    },

    @computed get active() {
        return map(Config.screen_ids, id => this.all.get(id));
    },

    reset() {
        Config.screen_ids.clear();
    },

});

export default Screens;
