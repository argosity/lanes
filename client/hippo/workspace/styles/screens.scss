.screen {
    padding: 5px;
    width: 100%;

    > .first-run {
        font-size: 1.3rem;
        li + li {
            margin-top: 1rem;
        }
        .help {
            margin-top: 2rem;
            p { text-align: right; }
        }
    }
}
