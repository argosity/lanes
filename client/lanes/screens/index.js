import { observable, autorun } from 'mobx';
import { map, uniq } from 'lodash';
import Group from './group';
import Config from '../config';
import Sync from '../models/sync';

import user from '../user';

const Screens = observable({

    all: observable.map(),

    active: observable.array(),

    groups: observable.map(),

    get activeGroups() {
        return uniq(map(this.active, s => Group.forId(s.group_id)));
    },

    refresh() {
        Sync.perform(`${Config.api_path}/lanes/screens`).then(({ data: { screens: screenIds } }) => {
            this.active.replace(map(screenIds, id => this.all.get(id)));
        });
    },

});
export default Screens;

let previousLoggedIn = null;
autorun(() => {
    if (user.isLoggedIn !== previousLoggedIn){
        previousLoggedIn = user.isLoggedIn;
        Screens.refresh();
    }
});
