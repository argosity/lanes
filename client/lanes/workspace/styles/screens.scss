.screen {
    padding: 5px;
    //  overflow: hidden;
    width: 100%;
    .toolbar {
        margin-top: 10px;
    }

    .toolbar{
        @extend .navbar;
        @extend .navbar-default;

        .btn {
            @extend .btn-default;
            @extend .navbar-btn;
        }

        .arrow {
            border-style: solid;

            border-left-color: transparent;
            border-right-color: transparent;

            border-bottom-color: $popover-title-bg;
            border-top-color: $popover-title-bg !important; 

            border-width: 0 9px 9px;
            height: 0;
            left: 7px;
            margin-top: 10px;
            opacity: 1;
            position: absolute;
            top: -18px;
            transition: all 0.25s ease 0s;
            width: 0;
            z-index: 10;
        }
        .navbar-left {
            float: left;
        }
        .navbar-right{
            float: right;
            .arrow {
                left: unset;
                right: 10px;
            }
        }

        .dropdown-toggle {
            .badge{
                padding: 2px 2px 0;
                position: relative;
                font-size: 70%;
                top: -6px;
                right: 5px;
            }
            position: relative;
            padding-right: 2px;
            i {
                /* position: absolute; */
                /* left: 0px; */
            }
        }

    }

}
