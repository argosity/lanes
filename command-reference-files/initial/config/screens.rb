# An example of how to define a screen group
# Hippo::Screen.define_group 'appy-app-screens' do | group |
#     group.title       = "AppyApp Screens"
#     group.description = "Screens relating to AppyApp"
#     group.icon        = "heart"
# end

Hippo::Screen.for_extension 'appy-app' do | screens |

end
