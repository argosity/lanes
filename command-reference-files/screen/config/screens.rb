# An example of how to define a screen group
# Hippo::Screen.define_group 'appy-app-screens' do | group |
#     group.title       = "AppyApp Screens"
#     group.description = "Screens relating to AppyApp"
#     group.icon        = "heart"
# end

Hippo::Screen.for_extension 'appy-app' do | screens |
    screens.define "ready-set-go" do | screen |
        screen.title       = "Ready Set Go"
        screen.description = ""
        screen.icon        = ""
        screen.group_id    = ""
        screen.model_class = ""
        screen.view_class  = "ReadySetGo"
    end

end
