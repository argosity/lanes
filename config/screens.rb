Lanes::Screen.define_group "system-settings" do | group |
    group.title       = "Settings"
    group.description = "Modify system settings"
    group.icon        = "wrench"
end

Lanes::Screen.for_extension 'lanes' do | screens |

    # screens.define "user-preferences" do | screen |
    #     screen.title       = "User Preferences"
    #     screen.description = "Modify User Profile"
    #     screen.icon        = "user-secret"
    #     screen.group_id    = "system-settings"
    #     screen.model_class = "User"
    #     screen.view_class  = "UserPreferences"
    #     screen.asset  = 'user'
    # end
    screens.define "user-management" do | screen |
        screen.title       = "User Management"
        screen.description = "Add/Remove/Modify User Accounts"
        screen.icon        = "group"
        screen.group_id    = "system-settings"
        screen.model_class = "SystemSettings"
        screen.view_class  = "UserManagement"
        screen.asset       = 'lanes/access/screens/user-management'
    end

    screens.define "system-settings" do | screen |
        screen.title       = "System Settings"
        screen.description = "Update System Settings"
        screen.icon        = "cogs"
        screen.group_id    = "system-settings"
        screen.model_class = "SystemSettings"
        screen.asset       = 'lanes/screens/system-settings'
    end

    # screens.define "workspace-layout" do | screen |
    #     screen.url_prefix  = 'lanes/workspace'
    #     screen.js  = 'index.jsx'
    #     # screen.css = 'styles.scss'
    # end
end
