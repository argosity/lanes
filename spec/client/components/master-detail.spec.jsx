import React from 'react';

import { Snapshot } from 'hippo/testing/screens';
import MasterDetail from 'hippo/components/master-detail';

function MasterComp() { return <h1>Hi, i am master</h1>; }
function DetailComp() { return <h1>Hi, i am detail</h1>; }

describe('Master Detail Component', () => {
    it('renders and matches snapshot', () => {
        expect(Snapshot(
            <MasterDetail
                master={<MasterComp />}
                detail={<DetailComp />}
            />)).toMatchSnapshot();
    });

    it('renders and then can add detail', () => {
        const md = shallow(<MasterDetail master={<MasterComp />} />);
        expect(md).not.toHaveRendered('.has-detail');
        md.setProps({ detail: <DetailComp /> });
        expect(md).toHaveRendered('.has-detail');
    });
});
