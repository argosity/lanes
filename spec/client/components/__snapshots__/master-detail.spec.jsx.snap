// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master Detail Component renders and matches snapshot 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="master-detail-wrapper has-detail"
  >
    <div
      className="master"
    >
      <h1>
        Hi, i am master
      </h1>
    </div>
    <div
      className="detail"
    >
      <h1>
        Hi, i am detail
      </h1>
    </div>
  </div>
</div>
`;
